### PR TITLE
[GLUTEN-2346][VL] Move Velox-Substrait plan conversion to Gluten cpp

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

To simply the process of Velox rebase and enhance code maintainability, this PR moves the Velox-Substrait plan conversion from Velox to Gluten cpp.

(Fixes: \#2346)

## How was this patch tested?

under test

### CODE REVIEW

File: cpp/velox/tests/VeloxToSubstraitTypeTest.cc [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (winningsix) [inline cpp/velox/tests/VeloxToSubstraitTypeTest.cc:2]: Question: Hmm, should License header add Intel's as well since moving under Intel's repo?

Comment (rui-mo) [reply cpp/velox/tests/VeloxToSubstraitTypeTest.cc]: Thanks for pointing out. Fixed.

**Review by zhejiangxiaomai — APPROVED** [review]
Thanks.

**Review by PHILO-HE — APPROVED** [review]

**Comment by Yohahaha** [comment]
Hi @rui-mo,  does that mean we can exclude build substrait in oap/velox?

**Comment by rui-mo** [comment]
> Hi @rui-mo, does that mean we can exclude build substrait in oap/velox?

@Yohahaha Yes, that's right. @zhejiangxiaomai will help remove the compilation of Substrait module in next rebase. All local changes on Substrait will also be remove from oap/velox.

**Comment by zhejiangxiaomai** [comment]
https://github.com/oap-project/velox/pull/366 remove the compilation of Substrait module

**Pull request merged**